src/act.drive.cpp

```diff
@@ -1270,8 +1270,8 @@ ACMD(do_driveby)
   }
 
   if ((wielded = GET_EQ(ch, WEAR_WIELD))) {
-    if (!(GET_OBJ_VAL(wielded, 4) == SKILL_PISTOLS || GET_OBJ_VAL(wielded, 4) == SKILL_SMG)) {
-      send_to_char(ch, "You can only do a driveby with a pistol or SMG.\r\n");
+    if (!(GET_OBJ_VAL(wielded, 4) == SKILL_PISTOLS || GET_OBJ_VAL(wielded, 4) == SKILL_SMG || ((0 <= GET_OBJ_VAL(wielded, 3) <= 4)  && ch->in_veh->type = VEH_BIKE)))  {
```

Comment:
This logic hasn't been touched in a dog's age. It probably needs to include IS_GUN for the weapon, and should have the magic number values removed (see utils.hpp for a list of GET_WEAPON_X calls to replace the GET_OBJ_VAL calls with).

Also, this should only allow doing drivebys with one-handed weapons.

Reply:
I want to be able to lance people or cut them in half with a ridiculously oversized sword. Also, this leaves open shooting people with one-handed shotguns.

Bikes are objectively worse than cars in pretty much all circumstances.